articles/user-guide/language/types.md

```diff
@@ -10,112 +10,115 @@ uid: microsoft.quantum.guide.types
+  Use the `string` type to report messages to a classical host in the case of an error or diagnostic event.
+- The `Unit` type can have only one value, `()`. 
+  Use this type to indicate that a Q# function or operation returns no information. 
 - The `Qubit` type represents a quantum bit or qubit.
-   `Qubit`s are opaque to the user; the only operation possible with them, other than passing them to another operation, is to test for identity (equality).
-   Ultimately, actions on `Qubit`s are implemented by calling intrinsic instructions on a quantum processor (or a simulation thereof).
+   `Qubit`s are opaque to the user. The only operation possible with them, other than passing them to another operation, is to test for identity (equality).
+   Ultimately, you implement actions on `Qubit`s by calling intrinsic instructions on a quantum processor (or a quantum simulator).
 - The `Pauli` type represents one of the four single-qubit Pauli operators.
-   This type is used to denote the base operation for rotations and to specify the observable being measured.
-   This is an enumerated type that has four possible values: `PauliI`, `PauliX`, `PauliY`, and `PauliZ`, which are constants of type `Pauli`.
+   Use this type to denote the base operation for rotations and to specify the observable being measured.
+   It is an enumerated type that has four possible values: `PauliI`, `PauliX`, `PauliY`, and `PauliZ`, which are constants of type `Pauli`.
 - The `Result` type represents the result of a measurement.
-   This is an enumerated type with two possible values: `One` and `Zero`, which are constants of type `Result`.
-   `Zero` indicates that the +1 eigenvalue was measured; `One` indicates the -1 eigenvalue.
+   It is an enumerated type with two possible values: `One` and `Zero`, which are constants of type `Result`.
+   `Zero` indicates that the +1 eigenvalue was measured; `One` indicates the -1 eigenvalue was measured.
 
 The constants `true`, `false`, `PauliI`, `PauliX`, `PauliY`, `PauliZ`, `One`, and `Zero` are all reserved symbols in Q#.
 
 ## Array Types
 
-Given any valid Q# type `'T`, there is a type that represents an array of values of type `'T`.
-This array type is represented as `'T[]`; for example, `Qubit[]` or `Int[][]`.
-For instance, a collection of integers is denoted `Int[]`, while an array of arrays of `(Bool, Pauli)` values is denoted `(Bool, Pauli)[][]`.
+* For any valid Q# type, there is a type that represents an array of values of that type.
+    For example, `Qubit[]` and `(Bool, Pauli)[]` represent arrays of `Qubit` values and `(Bool, Pauli)` tuple values.
 
-In the second example, note that this represents a potentially jagged array of arrays, and not a rectangular two-dimensional array.
-Q# does not provide support for rectangular multi-dimensional arrays.
+* An array of arrays is also valid. For example, an array of integer arrays is denoted `Int[][]`.
```

Comment:
Good catch. I kept the note because we are pointing out something that Q# doesn't support, but I clarified the examples leading up to it so the flow should make sense.